lighthouse-core/test/lib/sentry-test.js

```diff
@@ -0,0 +1,135 @@
+ */
+'use strict';
+
+jest.mock('raven');
+
+const raven = require('raven');
+const Sentry = require('../../lib/sentry');
+
+/* eslint-env jest */
+
+describe('Sentry', () => {
+  let configPayload;
+  let originalSentry;
+  let originalMathRandom;
+  let ravenConfigFn;
+  let ravenMergeContextFn;
+  let ravenCaptureExceptionFn;
+  let mathRandomFn;
+
+  beforeEach(() => {
+    configPayload = {
+      url: 'http://example.com',
+      flags: {enableErrorReporting: true},
+      environmentData: {},
+    };
+
+    originalSentry = {...Sentry};
+    originalMathRandom = Math.random;
+    ravenConfigFn = jest.fn().mockReturnValue({install: jest.fn()});
+    ravenMergeContextFn = jest.fn();
+    ravenCaptureExceptionFn = jest.fn().mockImplementation((err, opts, cb) => cb());
+    mathRandomFn = jest.fn().mockReturnValue(0.001);
+
+    Math.random = mathRandomFn;
+    raven.config = ravenConfigFn;
+    raven.mergeContext = ravenMergeContextFn;
+    raven.captureException = ravenCaptureExceptionFn;
+  });
+
+  afterEach(() => {
+    // Reset the methods on the Sentry object
+    Object.assign(Sentry, originalSentry);
+    Math.random = originalMathRandom;
+  });
+
+  describe('.init', () => {
+    it('should noop when !enableErrorReporting', () => {
+      Sentry.init({url: 'http://example.com', flags: {}});
+      expect(ravenConfigFn).not.toHaveBeenCalled();
+      Sentry.init({url: 'http://example.com', flags: {enableErrorReporting: false}});
+      expect(ravenConfigFn).not.toHaveBeenCalled();
+    });
+
+    it('should noop when not picked for sampling', () => {
+      mathRandomFn.mockReturnValue(0.5);
+      Sentry.init({url: 'http://example.com', flags: {enableErrorReporting: true}});
+      expect(ravenConfigFn).not.toHaveBeenCalled();
+    });
+
+    it('should initialize the raven client when enableErrorReporting', () => {
+      Sentry.init({
+        url: 'http://example.com',
+        flags: {
+          enableErrorReporting: true,
+          emulatedFormFactor: 'desktop',
+          throttlingMethod: 'devtools',
+        },
+        environmentData: {},
+      });
+
+      expect(ravenConfigFn).toHaveBeenCalled();
+      expect(ravenMergeContextFn).toHaveBeenCalled();
+      expect(ravenMergeContextFn.mock.calls[0][0]).toEqual({
+        extra: {
+          url: 'http://example.com',
+          deviceEmulation: true,
+          emulatedFormFactor: 'desktop',
+          throttlingMethod: 'devtools',
+        },
+      });
+    });
+  });
+
+  describe('.captureException', () => {
+    it('should forward exceptions to raven client', async () => {
+      Sentry.init(configPayload);
+      const error = new Error('oops');
+      await Sentry.captureException(error);
+
+      expect(ravenCaptureExceptionFn).toHaveBeenCalled();
+      expect(ravenCaptureExceptionFn.mock.calls[0][0]).toBe(error);
+    });
+
+    it('should skip expected errors', async () => {
+      Sentry.init(configPayload);
```

Comment:
what is an example of an expected error?

Reply:
An audit that fails because its artifact errored. In that case we'd essentially be double reporting. The real error is the one that failed in the gatherer, so we just report that one.

Reply:
https://github.com/GoogleChrome/lighthouse/blob/3d5037ade1b0b62b6bfe89642ba0e20904809df2/lighthouse-core/runner.js#L269-L274

Reply:
@Hoten it's pretty much only for controlling error reporting to Sentry. Basically it means this is an error, but we expect it often enough and it might be something the user needs to fix (or in the example @patrickhulce linked, we already reported it when it was a gatherer error, no reason to report it again).